.github/workflows/test.yml

```diff
@@ -55,11 +70,24 @@ jobs:
           name: OpenAssetIO Build
           path: ${{ github.workspace }}/openassetio
 
+      - name: Download wheels from commit OpenAssetIO main
```

Comment:
[consider] Not sure about this now. These tests should be using the "OpenAssetIO Build" step, above, so shouldn't need the `openassetio` wheel :thinking:

Reply:
Possibly by removing the wheel we'd get import errors as pytest scans the tests directory, but otherwise the wheel won't be used for building/running the CTest tests.

Perhaps the correct thing to do in that case would be to set `PYTHONPATH` to the `site-packages` directory in the built artifact, rather than relying on the wheel.

Reply:
Ahh yes - good spot - def +1 on extending `PYTHONPATH`

Reply:
(we should pin the checkout to a release version tag in future, to match the PyPI wheel)

Reply:
So this was a rabbithole.

Long story short, extending `PYTHONPATH` didn't work, and the `pytest` invocation fails to find the `openassetio` module. I don't know why, both David and myself looked at it and I put a cavalcade of debug output in trying to figure it out.
What did work in the end was just _setting_ `PYTHONPATH` though, as we don't need to actually execute any python, just have `pytest` be happy about it.

`PYTHONPATH` _is_ modified before/after the call into `pytest`, with some additional paths prepended, (the paths the test code is in mostly,) so perhaps that's causing the issue although I can't see why. Nonetheless it's not worth getting any further sidetracked in my opinion.

https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/51/commits/1244958266a0f6ccd81539ee83e40ff3aa707e17

Reply:
Great work figuring it out. Looks much neater to my eyes.